app/javascript/controllers/transcript_controller.js

```diff
@@ -16,21 +16,63 @@ export default class extends Controller {
   load() {
     if (this.loaded)
       return
-    const cues = this.textTrackCues().map((cue) => this.buildCue(cue))
-    this.outletTarget.innerHTML = cues.join('')
     this.revealButton()
+    this.setupTranscriptLanguageSwitching()
+    this.renderCues()
     this.loaded = true
-    this.setupTranscriptScroll()
   }
 
-  textTrackCues() {
-    const tracks = this.player.textTracks_.tracks_
-    if (!tracks)
-      return []
-    const track = tracks.find((track) => track.kind === 'captions' )
-    if (!track)
-      return []
-    return track.cues.cues_
+  get captionTracks() {
+    const tracks = this.player.textTracks_?.tracks_
+    if (!tracks) return []
+
+    return tracks.filter(track => track.kind === 'captions')
+  }
+
+  get cuesByLanguage() {
+    const cues = {}
+    if (this.captionTracks.length == 0) return cues
+
+    this.captionTracks.forEach(track => {
+      const list = track.cues.cues_
+      const cueStartTimes = list.length === 0 ? undefined : list.map((cue) => cue.startTime)
+
+      cues[track.language] = {
+        list,
+        cueStartTimes,
+        minStartTime: list.length === 0 ? 0 : Math.min.apply(Math, cueStartTimes),
+        lastCueEndTime: list.length === 0 ? 0 : Math.max.apply(Math, list.map((cue) => cue.endTime)),
+        asHtml: list.map((cue) => this.buildCue(cue)).join('')
+      }
+    })
+
+    return cues
+  }
+
+  currentCues() {
+    return this.selectedLanguage ?
+      this.cuesByLanguage[this.selectedLanguage] :
+      Object.values(this.cuesByLanguage)[0]
```

Comment:
This allows us to switch out the cues based on the selected language, defaulting to the first caption track (whatever the server gave us) since there's no selected language when the page loads.